### DISCUSSION TITLE
[Remix] Error: ENOENT: no such file or directory, copyfile '/var/task/node_modules/@vercel/remix/server-node.mjs' -> '/vercel/path0/build/server-node.mjs'

### DISCUSSION THREAD

**Discussion by jdeniau** [vercel/community#1631] [Help] [answered]
### Summary

When trying to update my blog under remix, without changing anything but HTML, I got the following error 

```
Error: ENOENT: no such file or directory, copyfile '/var/task/node_modules/@vercel/remix/server-node.mjs' -> '/vercel/path0/build/server-node.mjs'
```

I found a similar issue here https://github.com/orgs/vercel/discussions/1282 

Changing the environment variable to `VERCEL_CLI_VERSION=vercel@28.11.1` did fix the issue, but #1282 says that a fix has already been pushed and should not be needed.

Thank you

**Comment by TooTallNate** [comment, marked as answer]
Can you share what's in your `remix.config.js` file please?

**Reply by jdeniau** [reply]
Yes of course :

```
/**
 * @type {import('@remix-run/dev/config').AppConfig}
 */
module.exports = {
  appDirectory: "app",
  assetsBuildDirectory: "public/build",
  publicPath: "/build/",
  serverBuildDirectory: "api/_build",
  ignoredRouteFiles: [".*"]
};tve
```

**Reply by WesleyYue** [reply]
I also see this issue intermittently. Hitting redeploy on the dashboard fixes it for me.

Vercel CLI 28.16.6
```
// remix.config.js
const path = require("node:path");
const { flatRoutes } = require("remix-flat-routes");

/** @type {import('@remix-run/dev').AppConfig} */
module.exports = {
  future: {
    unstable_tailwind: true,
  },
  serverDependenciesToBundle: ["p-debounce"],
  ignoredRouteFiles: ["**/*", "**/.*"],
  routes: async (defineRoutes) => {
    return flatRoutes("routes", defineRoutes, {
      appDir: path.resolve(__dirname, "app"),
      ignoredRouteFiles: ["**/*.{mp4,png,css}", "**/.*", "**/*.bak"],
    });
  },
};

```

**Reply by TooTallNate** [reply]
Thanks for that. I haven't been able to reproduce myself so far. Can you also share some deployment URLs where this happened?

**Reply by jdeniau** [reply]
A vercel url of a failed build : 

[julien-deniau-miwshl94w-jdeniau.vercel.app](https://julien-deniau-miwshl94w-jdeniau.vercel.app/)

Urls that did succeeded with the ENV fix : 

[julien.deniau.me](http://julien.deniau.me/)
[julien-deniau-me-git-main-jdeniau.vercel.app](https://julien-deniau-me-git-main-jdeniau.vercel.app/)
[julien-deniau-8ce6q0lx7-jdeniau.vercel.app](https://julien-deniau-8ce6q0lx7-jdeniau.vercel.app/)

**Reply by WesleyYue** [reply]
Failed build: https://vercel.com/double-automation/double/HeG6hquHoZYk2NgUGioUkWcnr5pQ
Redeploy that succeeded: https://vercel.com/double-automation/double/DrxB9sA3Lzrj74rX9KJ5ud4E8puM

**Reply by TooTallNate** [reply]
@jdeniau I'm still debugging the underlying issue, but it's related to using an older version of Remix. I was able to get your site working [by updating to the latest version](https://github.com/jdeniau/julien.deniau.me/pull/9).

@WesleyYue I can't see the source code for your repo, but I'm willing to bet that it's a misconfiguration of Turbo. The `build` directory would need to be configured as an `output` of your app so that it gets properly cached.

**Reply by WesleyYue** [reply]
Ah, yep, that was it, thanks!

**Reply by jdeniau** [reply]
@TooTallNate Thanks for the PR, you are right, it is now fixed in the latest version.

Sorry I should have done that myself before !

**Comment by jdeniau** [comment]
Closing as both bugs are resolved

**Reply by jdeniau** [reply]
I can not close this as it's a discussion and not an issue 😅

**Reply by amyegan** [reply]
I marked the earlier thread as the "answer" since the PR was shared there. That's as good as "closed" for a discussion 😄
